### ISSUE TITLE
better support for metadata i/o

### ISSUE THREAD

**Issue by gidden** [IAMconsortium/pyam#220]
At present, the user has to keep track separately of their data and metadata for both output writing and input reading. It would be nice to have standard methods to facilitate this.

For example, at the moment one has to do the following for writing both datasets

```
df.data.to_csv('sr15_data.csv', index=False)
df.meta.to_csv('sr15_meta.csv', index=True)
```

and the following for reading both:

```
df = pyam.IamDataFrame('./sr15_data.csv')
mf = pd.read_csv('./sr15_meta.csv', index_col=[0, 1])
for col in mf:
    df.set_meta(mf[col])
```

**Comment by gidden** [comment]
only assigning to start discussion =)

**Comment by danielhuppmann** [comment]
good call @gidden!

I'd suggest the following API extensions:

## Excel

If initialising from an `xlsx` file, look for a sheet `meta` and import if available.

Extend `df.to_excel()` with an additional kwarg `include_meta` (default `True`). 

## csv

If initialising from a `csv` file, we could have a second kwarg `meta` that has the name of the `meta` csv file.

Extend `df.to_csv()` with an optional `meta_path` kwarg.

**Comment by danielhuppmann** [comment]
closed by #295 (for `xlsx`) - reopen if this requires more discussion
